src/Utils/util.ts

```diff
@@ -93,4 +94,15 @@ export function earlierDateOrTimeToday(timestamp: string): string {
     const dialogTime = moment(timestamp)
     const isDialogCreatedToday = dialogTime.diff(endOfYesterday) >= 0
     return dialogTime.format(isDialogCreatedToday ? 'LTS' : 'L')
+}
+
+export function isNewActionUnique(newAction: CLM.ActionBase, actions: CLM.ActionBase[]): boolean {
+    const needle = normalizeActionAndStringify(newAction)
+    const haystack = actions.map(action => normalizeActionAndStringify(action))
+    return haystack.findIndex(straw => straw === needle) < 0
```

Comment:
use `some` instead of `findIndex`.  You're not looking for an index your trying to see if anything is equal

Reply:
Good suggestion as some() is the intent. I'd considered the index at a later date should the UI offer to flag the matching action